QueryBuilder/Dynamic/Statements/WhereBaseStatement.cs

```diff
@@ -0,0 +1,69 @@
+// Licensed under the MIT License.
+
+namespace Microsoft.DigitalWorkplace.DigitalTwins.QueryBuilder.Dynamic.Statements
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics.CodeAnalysis;
+    using Microsoft.DigitalWorkplace.DigitalTwins.QueryBuilder.Common.Clauses;
+    using Microsoft.DigitalWorkplace.DigitalTwins.QueryBuilder.Common.Helpers;
+
+    /// <summary>
+    /// An abstract class that provides the base layer of methods that are common between
+    /// WHERE statement classes that are applicable to either Twins or Relationships.
+    /// </summary>
+    /// <typeparam name="TWhereStatement">The type of WHERE statement supported.</typeparam>
+    public abstract class WhereBaseStatement<TWhereStatement> where TWhereStatement : WhereBaseStatement<TWhereStatement>
+    {
+        /// <summary>
+        /// The alias for the current WHERE statement.
+        /// </summary>
+        protected readonly string Alias;
+        internal readonly WhereClause WhereClause;
+        internal readonly IEnumerable<JoinOptions> JoinOptions;
+
+        [ExcludeFromCodeCoverage]
+        internal WhereBaseStatement()
+        {
+        }
+
+        internal WhereBaseStatement(IEnumerable<JoinOptions> joinOptions, WhereClause clause, string alias) : this(clause, alias)
+        {
+            JoinOptions = joinOptions;
+        }
+
+        internal WhereBaseStatement(WhereClause whereClause, string alias)
+        {
+            WhereClause = whereClause;
+            Alias = alias;
+        }
+
+        /// <summary>
+        /// Wraps a given set of WHERE conditions in parenthesis to establish a precedence.
+        /// </summary>
+        /// <param name="nested">The functional logic of the WHERE statement containing one or more WHERE conditions.</param>
+        /// <returns>A conjunction class that supports appending more conditions to the WHERE statements via OR or AND terms.</returns>
+        public CompoundWhereStatement<TWhereStatement> Precedence(Func<TWhereStatement, CompoundWhereStatement<TWhereStatement>> nested)
+        {
+            var statement = WhereStatementFactory.CreateInstance<TWhereStatement>(JoinOptions, new WhereClause(), Alias);
+            var n = nested.Invoke(statement);
+            WhereClause.AddCondition($"({n.WhereClause.GetConditionsText()})");
+            return new CompoundWhereStatement<TWhereStatement>(JoinOptions, WhereClause, Alias);
+        }
+
+        /// <summary>
+        /// Prepends a NOT term to one or more WHERE clauses.
+        /// </summary>
+        /// <param name="nested">The functional logic of the WHERE statement containing one or more WHERE conditions.</param>
+        /// <returns>A conjunction class that supports appending more conditions to the WHERE statements via OR or AND terms.</returns>
+        public CompoundWhereStatement<TWhereStatement> Not(Func<TWhereStatement, CompoundWhereStatement<TWhereStatement>> nested)
```

Comment:
1) It should be clear because the documentation doesn't say that it does. Documentation is meant to explain what a method _does_, not what it _does not_ do.
2) This follows the same concept of `.And()` and `.Or()` the only difference between this and those is that I've provided a parameterless `.And()` and one that takes conditions (same with Or). I felt it was clearer to only add parentheses via the explicit use of `.Precedence`. That being said, I can see how one might guess `.And(conditions)` might establish precedence whereas `.And()` does not - especially considering the Typed builder implicitly adds parentheses all over the place.. (something I think should be changed)

I think the options here are to either having `.And(conditions)`, `.Or(conditions)`, and `.Not(conditions)` establish precedence implicitly and having `.And()` and `.Or()` remain the same; then, add a parameterless `.Not()`
or, remove `.And(conditions)`, `.Or(conditions)` and `.Not(conditions)` in favor of the parameterless counterparts and ensure precedence is only established with `.Precedence`. Personally, I'd prefer the latter as I'm not a fan of implied precedence.